app/lib/features/attachments/widgets/reference_attachment_item.dart

```diff
@@ -25,74 +35,178 @@ class ReferenceAttachmentItem extends ConsumerWidget {
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final lang = L10n.of(context);
-    final containerColor = Theme.of(context).colorScheme.surface;
-    final eventId = attachment.attachmentIdStr();
-    final roomId = attachment.roomIdStr();
-
     return Container(
       decoration: BoxDecoration(
-        color: containerColor,
+        color: Theme.of(context).colorScheme.surface,
         borderRadius: BorderRadius.circular(8),
         border: Border.all(color: Theme.of(context).unselectedWidgetColor),
       ),
       child: ListTile(
-        leading: PhosphorIcon(PhosphorIconsThin.tagChevron),
-        title: title(context),
-        trailing: Row(
-          mainAxisSize: MainAxisSize.min,
-          children: [
-            if (canEdit)
-              PopupMenuButton<String>(
-                key: const Key('attachment-item-menu-options'),
-                icon: const Icon(Icons.more_vert),
-                itemBuilder: (context) => [
-                  PopupMenuItem<String>(
-                    key: const Key('attachment-delete'),
-                    onTap: () {
-                      openRedactContentDialog(
-                        context,
-                        eventId: eventId,
-                        roomId: roomId,
-                        title: lang.deleteAttachment,
-                        description:
-                            lang.areYouSureYouWantToRemoveAttachmentFromPin,
-                        isSpace: true,
-                      );
-                    },
-                    child: Text(
-                      lang.delete,
-                      style: TextStyle(
-                        color: Theme.of(context).colorScheme.error,
-                      ),
-                    ),
-                  ),
-                ],
-              ),
-          ],
-        ),
+        onTap: () => onTapRefAttachment(context, ref),
+        leading: refAttachmentIcons(ref),
+        title: refAttachmentTitle(ref),
+        subtitle: refAttachmentSubTitle(),
+        trailing: actionMenu(context),
       ),
     );
   }
 
-  Widget title(BuildContext context) {
-    final title = refDetails.title();
-    return Column(
-      crossAxisAlignment: CrossAxisAlignment.start,
+  Widget refAttachmentTitle(WidgetRef ref) {
+    final defaultTitle = Text(refDetails.title().toString());
```

Comment:
can still technically be `null` and should be dealt with... 
```suggestion
    final defaultTitle = Text(refDetails.title() ?? L10n.of(context).unknown);
```

 (or something)